pkgs/build-support/go/module.nix

```diff
@@ -63,6 +63,11 @@ let
   GO111MODULE = "on";
   GOTOOLCHAIN = "local";
 
+  # When building PIE binaries for Linux, we must set CGO_ENABLED and enable external linking.
+  # Otherwise the emitted binary still depends on ld.so but at wrong path
+  linuxGnuPie = builtins.elem "pie" stdenv.cc.bintools.defaultHardeningFlags
```

Comment:
Does this respect hardening{Disable,Enable} that could be set in a package?

Reply:
Those are usually set at runtime in shell. There is currently a check for musl that enables pie. Maybe it can also disable pie if it's not in NIX_HARDENING_ENABLE

Reply:
That being sad on musl we now would add the flag twice:

```
    if [[ $NIX_HARDENING_ENABLE =~ "pie" ]]; then
        export GOFLAGS="-buildmode=pie $GOFLAGS"
    fi
```

So maybe we should drop or adapt this.